site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -2331,6 +2355,165 @@ public function ajaxSaveGradedComponent($gradeable_id) {
         }
     }
 
+    /**
+     * @param string $gradeable_id
+     * @return JsonResponse|void
+     */
+    #[Route("/courses/{_semester}/{_course}/gradeable/{gradeable_id}/grading/graded_gradeable/open_component", methods: ["POST"])]
+    public function ajaxOpenGradedComponent($gradeable_id) {
+        $anon_id = $_POST['anon_id'] ?? '';
+        $component_id = $_POST['component_id'] ?? '';
+        $grader = $this->core->getUser();
+
+        // Get the gradeable
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+        if ($gradeable === false) {
+            return;
```

Comment:
It would be better to return a `JsonResponse::getErrorResponse()` or similar instead of exiting with no explanation.  Apply throughout.